### PR TITLE
Implemented IGH Oceanic View

### PR DESCRIPTION
**Pull request by jkrasting**
- Option to select 'land' or 'ocean' emphasis
- Requires an [update to the PROJ library](https://github.com/OSGeo/PROJ/pull/2226) and use of
  a new projection (imh_o)
- Ocean view works best with central_longitude=-160.

IGH Default View:

```
proj = ccrs.InterruptedGoodeHomolosine()
```

imh_default_view

IGH New Optional Ocean View:

```
proj = ccrs.InterruptedGoodeHomolosine(central_longitude=-160,emphasis='ocean')
```

imh_ocean_view

## Rationale

This update provides new functionality for oceanographic plots that was not previously possible with the land-only focus of the current IGH projection.

## Implications

None.

### CODE REVIEW

**Comment by jkrasting** [comment]
As US Government employee, this contribution is considered in the public domain.

**Comment by QuLogic** [comment]
Now that Proj 7.1.0 is out, we can add this. However, it will need a check that the version of Proj is new enough, like `EqualEarth`. It will also need an entry in the docs in `docs/make_projection.py` and `docs/source/crs/projections.rst`, and the what's new `docs/whats_new.rst`.

**Comment by jkrasting** [comment]
OK - sounds good.  What is the timeline for the next version release?

**Comment by QuLogic** [comment]
No specific plans, but probably not for a couple months.

**Comment by jkrasting** [comment]
@QuLogic - I added in a PROJ version check and updated the documentation.  I didn't see a `whats_new.rst` file for the an new version, however.

**Comment by jkrasting** [comment]
image

**Comment by QuLogic** [comment]
Can you rebase to get working CI, and ensure that tests are passing?

File: lib/cartopy/crs.py [hunk 1]
```diff
@@ -1930,17 +1930,59 @@ def transform_points(self, src_crs, x, y, z=None):
...
+            The central longitude. Defaults to 0.
+        globe: :class:`cartopy.crs.Globe`, optional
+            If omitted, a default globe is created.
+        emphasis: optional
+            Options "land" and "ocean" are available. Defaults to "land"
+
+        """
+        if emphasis == 'land':
+            proj4_params = [('proj', 'igh'), ('lon_0', central_longitude)]
+            super().__init__(proj4_params,globe=globe)
```
Comment (jkrasting) [inline lib/cartopy/crs.py:1960]:
```suggestion
            super().__init__(proj4_params, globe=globe)
```

File: lib/cartopy/crs.py [hunk 2]
```diff
@@ -1930,17 +1930,59 @@ def transform_points(self, src_crs, x, y, z=None):
...
+
+        elif emphasis == 'ocean':
+            if PROJ4_VERSION < (7, 1, 0):
+                _proj_ver = '.'.join(str(v) for v in PROJ4_VERSION)
+                raise ValueError('The Interrupted Goode Homolosine ocean '
+                                 'projection requires Proj version 7.1.0, '
+                                 'but you are using ' + _proj_ver)
+            proj4_params = [('proj', 'igh_o'), ('lon_0', central_longitude)]
+            super().__init__(proj4_params,globe=globe)
```
Comment (jkrasting) [inline lib/cartopy/crs.py:1969]:
```suggestion
            super().__init__(proj4_params, globe=globe)
```

**Comment by jkrasting** [comment]
I rebased against the current `master` branch @QuLogic.  I also squashed into a single commit.

**Comment by jkrasting** [comment]
@QuLogic - I don't really understand the tests well enough to understand how this is failing.  Are you able to help me?

File: docs/source/whats_new.rst [hunk 3]
```diff
@@ -10,6 +10,9 @@ For a full list of included Pull Requests and closed Issues, please see the
 Features
 --------
 
+* John Krasting added the ability to use the oceanic view
```
Comment (greglucas) [inline docs/source/whats_new.rst:13]: This is for the previous release What's New. We generally make this while doing the release. You can add yourself to the contributors list now though.
https://github.com/SciTools/cartopy/blob/master/docs/source/contributors.rst

File: lib/cartopy/crs.py [hunk 4]
```diff
@@ -1930,17 +1930,59 @@ def transform_points(self, src_crs, x, y, z=None):
...
+        DOI: 10.1080/00045602509356949
+
+    A central_longitude value of -160 is recommended for the oceanic view.
+
+    """
+    def __init__(self, central_longitude=0, globe=None, emphasis='land'):
+        """
+        Parameters
+        ----------
+        central_longitude: optional
```
Comment (greglucas) [inline lib/cartopy/crs.py:1950]:
```suggestion
        central_longitude : float, optional
```
All of these should have a space after the name, then the type, followed by optional. See here for the parameter spec: https://numpydoc.readthedocs.io/en/latest/format.html#parameters

**Review by greglucas — COMMENTED** [review]
Looks like this is really close. The minimum version test is currently failing due to an old proj being used. You'll need to add a `pytest.skipif` to check versions for this, something similar to this:
https://github.com/SciTools/cartopy/blob/4ca93c1eefa86cb85177467407852b4636bb3e70/lib/cartopy/tests/mpl/test_mpl_integration.py#L243-L244
But probably within the parameterized function itself rather than as a decorator.

Do you want to add an image comparison for the new projection?

**Comment by jkrasting** [comment]
Thanks for your help @greglucas!  In regard to this suggestion:

> Do you want to add an image comparison for the new projection?

Did you mean add a figure in `lib/cartopy/tests/mpl/baseline_images/mpl/test_crs` or are you suggesting something different?

**Comment by greglucas** [comment]
Yes, I think that would be a good place for it. Just to make sure the image for the ocean view is coming out as you expect.

**Comment by jkrasting** [comment]
@greglucas - thanks for the feedback.  I think everything is addressed now.   This PR is blocked again from running the CI, but everything seems to check out OK on my fork:  https://github.com/jkrasting/cartopy/actions/runs/929096625

**Review by greglucas — APPROVED** [review]
I think this is a nice addition and good to go. I'll hold off merging for a few days in case anyone else has comments/suggestions on it.

**Comment by jkrasting** [comment]
> I think this is a nice addition and good to go. I'll hold off merging for a few days in case anyone else has comments/suggestions on it.

Thanks for your help @greglucas.  Looking forward to seeing this in a future release!

File: docs/source/crs/projections.rst [hunk 5]
```diff
@@ -274,10 +274,19 @@ InterruptedGoodeHomolosine
...
+                          projection=ccrs.InterruptedGoodeHomolosine(
+                              emphasis='land'))
+    ax1.coastlines(resolution='110m')
+    ax1.gridlines()
+
+    ax2 = fig.add_subplot(2, 1, 2,
+                          projection=ccrs.InterruptedGoodeHomolosine(
+                              central_longitude=-160, emphasis='ocean'))
+    ax2.coastlines(resolution='110m')
+    ax2.gridlines()
```
Comment (QuLogic) [inline docs/source/crs/projections.rst:289]: This file is generated by `make_projection.py`; you need to update that first.

Comment (jkrasting) [reply docs/source/crs/projections.rst:289]: Indeed, this code block is already in `make_projection.py`.

Comment (QuLogic) [reply docs/source/crs/projections.rst:289]: Hmm, right, don't know why I didn't see it before.

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Can you rebase to fix the conflicts?

**Comment by jkrasting** [comment]
Yes, just rebased now.

**Comment by jkrasting** [comment]
@greglucas, @QuLogic - looks like this needs a workflow approval for parts of the CI to run.

**Comment by QuLogic** [comment]
Thanks @jkrasting!

**Pull request merged**